alf/layers.py

```diff
@@ -404,6 +527,129 @@ def bias(self):
 
 
+@gin.configurable
+class ParallelConvTranspose2D(nn.Module):
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 kernel_size,
+                 n,
+                 activation=torch.relu_,
+                 strides=1,
+                 padding=0,
+                 use_bias=True,
+                 kernel_initializer=None,
+                 kernel_init_gain=1.0,
+                 bias_init_value=0.0):
+        """A 2D ParallelConvTranspose2D layer that's also responsible for
+        activation and customized weights initialization. An auto gain
+        calculation might depend on the activation following the conv layer.
+        Suggest using this wrapper module instead of ``nn.ConvTranspose2d``
+        if you really care about weight std after init.
+
+        Args:
+            in_channels (int): channels of the input image
+            out_channels (int): channels of the output image
+            kernel_size (int or tuple):
+            n (int): n indepedent ``ConvTranspose2D`` layer
+            activation (torch.nn.functional):
+            strides (int or tuple):
+            padding (int or tuple):
+            use_bias (bool):
+            kernel_initializer (Callable): initializer for the conv_trans layer.
+                If None is provided a variance_scaling_initializer with gain as
+                ``kernel_init_gain`` will be used.
+            kernel_init_gain (float): a scaling factor (gain) applied to the
+                std of kernel init distribution. It will be ignored if
+                ``kernel_initializer`` is not None.
+            bias_init_value (float): a constant
+        """
+        super(ParallelConvTranspose2D, self).__init__()
+        self._activation = activation
+        self._n = n
+        self._in_channels = in_channels
+        self._out_channels = out_channels
+        self._kernel_size = kernel_size
+        self._conv_trans2d = nn.ConvTranspose2d(
+            in_channels * n,
+            out_channels * n,
+            kernel_size,
+            groups=n,
+            stride=strides,
+            padding=padding,
+            bias=use_bias)
+
+        for i in range(n):
+            if kernel_initializer is None:
+                variance_scaling_init(
+                    self._conv_trans2d.weight.data[i * in_channels:(i + 1) *
+                                                   in_channels],
+                    gain=kernel_init_gain,
+                    nonlinearity=self._activation)
+            else:
+                kernel_initializer(
+                    self._conv_trans2d.weight.data[i * in_channels:(i + 1) *
+                                                   in_channels])
+
+        if use_bias:
+            nn.init.constant_(self._conv_trans2d.bias.data, bias_init_value)
+
+    def forward(self, img):
+        """Forward
+
+        Args:
+            img (torch.Tensor): with shape ``[B, C, H, W]``
```

Comment:
format; explain the difference between with n and without n

Reply:
Format fixed